### PR TITLE
Feature/reservation

### PR DESCRIPTION
**Pull request by aliatayee**
## Add reservation form
- Add reservation form
- Add rooms list dropdown

### CODE REVIEW

**Review by Chrissiku — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi @aliatayee 👋

Your project is complete! ✔️ 🏁

Finally, **you did it**! 🎉 👏. Your code was very professional 💯 and I can see you have a great chance to be a Great Software Engineer ⛑️.

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉. And Welcome to Microverse 👋.

> **Note:** You can now take a break even if short before you move on to the next project. Happy 😄 Coding 🖥️! and Have a nice day 🌞.

## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors
- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

## Thank you for your time 🙏 and call me if you need help or clarification ☎️
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wale-prog — APPROVED** [review]
# APPROVED!!!! 🎉 
## Hey @aliatayee 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!t 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/pages/add-reservation/ReserveForm.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { addNewReserve } from '../../../redux/features/reservations/mine';
+import { getRooms } from '../../../redux/roomsSlice';
+
+function ReserveForm() {
+    const dispatch = useDispatch();
+    const { data, loading, error } = useSelector((state) => state.rooms);
+
+    useEffect(() => {
+        dispatch(getRooms());
+    }, [dispatch]);
```
Comment (sntakirutimana72) [inline src/components/pages/add-reservation/ReserveForm.js:15]: - Kindly, complete all the expected behaviors when fetching available rooms bearing in mind that users can only reserve rooms that aren't currently reserved.

**Review by sntakirutimana72 — CHANGES_REQUESTED** [review]
Hi @aliatayee,

# Status: CHANGES REQUIRED

You've done an amazing job so far.
However, there are a few flaws you have to improve.

## Highlight! :green_circle: 

:heavy_check_mark:  Responsive design
:heavy_check_mark:  Good topography
:heavy_check_mark:  Good commit messages

## [REQUIRED] changes

- Your logic for fetching rooms from the `API` is incomplete, probably because your endpoint is not fully implemented. Kindly fix it. Kindly see the below comment for more details.

## [Optional] suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

-----

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack using this handle @sntakirutimana72.

**Review by safar1212 — APPROVED** [review]
# APPROVED!!!! 🎉
## Hey @aliatayee    😊 😃 

#### Great Job 😄 👏 You've done it. Nice completing this milestone 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sntakirutimana72 — APPROVED** [review]
@Hi @aliatayee,

# Status: APPROVED!! 🍾 

You've done an amazing job so far.
However, there are a few flaws you have to improve.

## Highlight! :green_circle: 

:heavy_check_mark:  Responsive design
:heavy_check_mark:  Good topography
:heavy_check_mark:  Good commit messages

## [Optional] suggestions

- Your logic for fetching rooms from the `API` is incomplete, probably because your endpoint is not fully implemented. Kindly fix it. Kindly see the below comment for more details.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

-----

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack using this handle @sntakirutimana72.

**Pull request merged**